src/batches/setBatchSegmentation.m

```diff
@@ -22,21 +22,38 @@
   % define SPM folder
   spmLocation = spm('dir');
 
-  % SAVE BIAS CORRECTED IMAGE
-  matlabbatch{end + 1}.spm.spatial.preproc.channel.vols(1) = ...
-      cfg_dep('Named File Selector: Anatomical(1) - Files', ...
-              substruct( ...
-                        '.', 'val', '{}', {opt.orderBatches.selectAnat}, ...
-                        '.', 'val', '{}', {1}, ...
-                        '.', 'val', '{}', {1}, ...
-                        '.', 'val', '{}', {1}), ...
-              substruct('.', 'files', '{}', {1}));
+  % save bias correction field = false
+  % save bias corrected image = true
+  matlabbatch{end + 1}.spm.spatial.preproc.channel.write = [false true];
+
+  if isfield(opt, 'orderBatches') && isfield(opt.orderBatches, 'selectAnat')
+
+    % SAVE BIAS CORRECTED IMAGE
+    matlabbatch{end}.spm.spatial.preproc.channel.vols(1) = ...
+        cfg_dep('Named File Selector: Anatomical(1) - Files', ...
+                substruct( ...
+                          '.', 'val', '{}', {opt.orderBatches.selectAnat}, ...
+                          '.', 'val', '{}', {1}, ...
+                          '.', 'val', '{}', {1}, ...
+                          '.', 'val', '{}', {1}), ...
+                substruct('.', 'files', '{}', {1}));
+  else
+
+    % in case a cell was given as input
+    if iscell(imageToSegment)
+      imageToSegment = char(imageToSegment);
+    end
+
+    % add all the images to segment
+    for iImg = 1:size(imageToSegment, 1)
+      file = validationInputFile([], deblank(imageToSegment(iImg, :)));
+      matlabbatch{end}.spm.spatial.preproc.channel.vols{iImg, 1} = file;
+    end
```

Comment:
Added a couple of things here:
- in case the users passes a cellstring as input, we covert it
- we make sure that the files we want to segment actually exist on the harddrive (better to fail now that at runtime)
- in case several files are passed to be segmented then we had all of them to the batch